clients/src/test/java/org/apache/kafka/clients/consumer/internals/PrototypeAsyncConsumerTest.java

```diff
@@ -16,90 +16,77 @@
+import org.junit.jupiter.api.AfterEach;
 import org.junit.jupiter.api.BeforeEach;
 import org.junit.jupiter.api.Test;
-import org.mockito.Mockito;
 
+import java.time.Duration;
 import java.util.HashMap;
 import java.util.Map;
-import java.util.Optional;
 
-import static java.util.Collections.singleton;
-import static org.apache.kafka.clients.consumer.ConsumerConfig.CLIENT_ID_CONFIG;
+import static org.apache.kafka.clients.consumer.ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG;
 import static org.apache.kafka.clients.consumer.ConsumerConfig.KEY_DESERIALIZER_CLASS_CONFIG;
 import static org.apache.kafka.clients.consumer.ConsumerConfig.VALUE_DESERIALIZER_CLASS_CONFIG;
-import static org.junit.jupiter.api.Assertions.assertEquals;
 import static org.junit.jupiter.api.Assertions.assertThrows;
 
 public class PrototypeAsyncConsumerTest {
-    private Map<String, Object> properties;
-    private SubscriptionState subscriptionState;
-    private MockTime time;
-    private LogContext logContext;
-    private Metrics metrics;
-    private ClusterResourceListeners clusterResourceListeners;
-    private Optional<String> groupId;
-    private String clientId;
-    private EventHandler eventHandler;
+
+    private Consumer<?, ?> consumer;
+    private Map<String, Object> consumerProps = new HashMap<>();
+
+    private final Time time = new MockTime();
 
     @BeforeEach
     public void setup() {
-        this.subscriptionState = Mockito.mock(SubscriptionState.class);
-        this.eventHandler = Mockito.mock(DefaultEventHandler.class);
-        this.logContext = new LogContext();
-        this.time = new MockTime();
-        this.metrics = new Metrics(time);
-        this.groupId = Optional.empty();
-        this.clientId = "client-1";
-        this.clusterResourceListeners = new ClusterResourceListeners();
-        this.properties = new HashMap<>();
-        this.properties.put(ConsumerConfig.BOOTSTRAP_SERVERS_CONFIG,
-                "localhost" +
-                ":9999");
-        this.properties.put(KEY_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
-        this.properties.put(VALUE_DESERIALIZER_CLASS_CONFIG, StringDeserializer.class);
-        this.properties.put(CLIENT_ID_CONFIG, "test-client");
+        injectConsumerConfigs();
+    }
+
+    @AfterEach
+    public void cleanup() {
+        if (consumer != null) {
+            consumer.close(Duration.ZERO);
+        }
     }
+
     @Test
-    public void testSubscription() {
-        this.subscriptionState =
-                new SubscriptionState(new LogContext(), OffsetResetStrategy.EARLIEST);
-        PrototypeAsyncConsumer<String, String> consumer =
-                setupConsumerWithDefault();
-        subscriptionState.subscribe(singleton("t1"),
-                new NoOpConsumerRebalanceListener());
-        assertEquals(1, consumer.subscription().size());
+    public void testBackgroundThreadRunning() {
+        consumer = newConsumer(time, new StringDeserializer(), new StringDeserializer());
```

Comment:
+1

Reply:
I'm thinking what's the best way to test that. As we don't have direct access to the background thread from the consumer.  I guess we could do that passively by trying to catch the exception, or maybe, actively, I should add a `public State state()` to the eventHandler?  to allow user to prob the background thread stat... I feel we need both, but WDYT? @rajinisivaram @guozhangwang

Reply:
maybe we don't need this afterall actually. I feel for the purpose of unit test, we should just try to test the API and method calls. The integration test in this PR is already kind of testing this (otherwise request won't come through)